### PR TITLE
Fix documentation/examples issues for async

### LINKED ISSUE #258: Deferring Asynchronous Work example from README does not match with example
In 
https://github.com/opentracing/opentracing-java/blob/master/README.md#deferring-asynchronous-work

The example has 3 steps:

1. Start a Span via either startManual or startActive(false) to prevent the Span from being finished upon Scope deactivation.
1. In the closure/Runnable/Future/etc itself, invoke tracer.scopeManager().activate(span, false) to re-activate the Span and get a new Scope, then deactivate() it when the Span is no longer active (or use try-with-resources for less typing).
1. In the closure/Runnable/Future/etc where the end of the task is reached, invoke `tracer.scopeManager().activate(span, true)` to re-activate the Span and have the new Scope close the Span automatically.

However, the code sample has only two steps:

```java
io.opentracing.Tracer tracer = ...;
...
// STEP 1 ABOVE: start the Scope/Span
try (Scope scope = tracer.buildSpan("ServiceHandlerSpan").startActive(false)) {
    ...
    final Span span = scope.span();
    doAsyncWork(new Runnable() {
        @Override
        public void run() {

            // STEP 2 ABOVE: reactivate the Span in the callback, passing true to
            // startActive() if/when the Span must be finished.
            try (Scope scope = tracer.scopeManager().activate(span, false)) {
                ...
            }
        }
    });
}
```

Since `tracer.scopeManager().activate(span, true)` is never called here, either the code sample or the text should be updated.

### PR DESCRIPTION
**Pull request by tylerbenson**
@wsargent, does this help clear things up?  In reality, there are several different ways this could be done, depending on the desired semantics and the design of the framework.

Closes #258.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,14 +114,14 @@ io.opentracing.Tracer tracer = ...;
...
     final Span span = scope.span();
     doAsyncWork(new Runnable() {
         @Override
         public void run() {
-
-            // STEP 2 ABOVE: reactivate the Span in the callback, passing true to
-            // startActive() if/when the Span must be finished.
-            try (Scope scope = tracer.scopeManager().activate(span, false)) {
+            // STEP 3 ABOVE: reactivate the Span on the new thread, passing true to
+            // startActive(). Span will be closed when try block completes.
```
Comment (wsargent) [inline README.md]: And then "finish" is like "span is completed and cannot be reactivated" and "activate(span, true)" says "after scopeManager.close() is called here, call span.finish() and you can't use this span after this point"?

What's the difference between "span will be closed" and "span will be finished"?

File: README.md [hunk 2]
```diff
@@ -102,8 +102,8 @@ The `"ServiceHandlerSpan"` is _active_ while it's running FunctionA and Function
 
 **The `ScopeManager` API makes it possible to fetch the `span()` in `FunctionA` and re-activate it in `FunctionB`.** Note that every `Tracer` contains a `ScopeManager`. These are the steps:
 
-1. Start a `Span` via either `startManual` or `startActive(false)` to prevent the `Span` from being finished upon `Scope` deactivation.
-2. In the closure/`Runnable`/`Future`/etc itself, invoke `tracer.scopeManager().activate(span, false)` to re-activate the `Span` and get a new `Scope`, then `deactivate()` it when the `Span` is no longer active (or use try-with-resources for less typing).
+1. Start a `Span` via either `start()` or `startActive(false)` to prevent the `Span` from being finished upon `Scope` deactivation.
```
Comment (wsargent) [inline README.md]: So I think there's some ambiguity here still.  

It seems like "start" and "startActive" are more like "create a new span" and "create a new span using the active span as a parent".  But it doesn't say whether the span is "active" or not when it's started or not.  I don't see know what Scope deactivation is, or how it relates to span activation and deactivation.

File: README.md [hunk 3]
```diff
@@ -102,8 +102,8 @@ The `"ServiceHandlerSpan"` is _active_ while it's running FunctionA and Function
...
+2. Pass the span along to the next thread.
```
Comment (wsargent) [inline README.md]: So passing a span to another thread can be done whether the span is active or inactive, but you cannot do things with the span until you call "activate" -- that does the work of saying "start the span (if it is not finished) and set it as the ActiveSpan in this thread"?

**Comment by akodali18** [comment]
@tylerbenson Is this issue still being worked on ?

**Comment by tylerbenson** [comment]
@akodali18 no.  I've moved onto other things.  I'll go ahead and close it.

**Pull request closed without merging**